consensus/test/member_management_test.go

```diff
@@ -0,0 +1,516 @@
+package test
+
+import (
+	"context"
+	"github.com/clearmatics/autonity/accounts/abi/bind"
+	"github.com/clearmatics/autonity/common"
+	"github.com/clearmatics/autonity/common/keygenerator"
+	"github.com/clearmatics/autonity/contracts/autonity"
+	"github.com/clearmatics/autonity/core"
+	"github.com/clearmatics/autonity/core/types"
+	"github.com/clearmatics/autonity/crypto"
+	"github.com/clearmatics/autonity/ethclient"
+	"github.com/clearmatics/autonity/p2p/enode"
+	"math/big"
+	"strconv"
+	"sync"
+	"testing"
+)
+
+/*
+  In this file, it create 4 test cases which have similar work flow base on the local e2e test framework's main flow.
+
+  First it setup an autontiy network by according to the genesis hook function, then from the specific chain height, it
+  start to issue transactions via the transaction hook function specified for the target node, for example in the
+  addValidatorHook, it issues transaction to call autonity contract via operator account to add a new validator.
+
+  Then the test case verify the output from its finalAssert hook function on the specified height of the blockchain, for
+  example the addValidatorCheckerHook checks if the new validator is presented in the white list, and its stake balance
+  checked too, and finally it checks the total stake supply after the membership updates.
+
+  for the other cases in the file: add stake_holder, participants, or remove user, they follow the same work flow and
+  some rules to check the outputs.
+*/
+
+func TestMemberManagementAddNewValidator(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+	once := sync.Once{}
+	// prepare chain operator
+	operatorKey, err := keygenerator.Next()
+	if err != nil {
+		t.Fatal(err)
+	}
+	operatorAddress := crypto.PubkeyToAddress(operatorKey.PublicKey)
+
+	newValidatorKey, err := keygenerator.Next()
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Lets use testify's assert and require instead. They help cut down the noise in tests,
```suggestion
	require.NoiError(t, err)
```

Reply:
Hi @Jason-Zhangxin-Chen, in my previous comment I meant lets use testify and assert for all of our assertions in these tests, not just the specific case I picked out.

Reply:
@Jason-Zhangxin-Chen I can still see some ocurrences of `t.Fatal`